### PR TITLE
Logging v2 part 3 strings

### PR DESCRIPTION
**Pull request by tiinanguyen**
This PR adds the following string files: `OfferingStrings.swift`, `PurchaseStrings.swift`, and `RestoreStrings.swift`.

Part 1: #426 
Part 2: #428 
Additional log methods: #422

### CODE REVIEW

**Review by tiinanguyen — COMMENTED** [review]
Only adding string files first. I also added a comment at the end of each string with the log method. Can I get a review on the names and respective logging method?

File: PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//  OfferingStrings.swift
+//  PurchasesCoreSwift
+//
+//  Created by Tina Nguyen on 12/11/20.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCOfferingStrings) public class OfferingStrings: NSObject {
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift:11]: not needed for this PR, but we should add a static instance of this class (and the others like it) to `Strings.swift`

File: PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    @objc public var cannot_find_product: String { "Could not find SKProduct for %@ \nThere is a problem with your configuration in App Store Connect. \nMore info here: http://errors.rev.cat/products-empty"} //appleWarning
+    @objc public var completion_handler_waiting_on_products: String { "%lu completion handlers waiting on products" } //debug
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift:13]: `completion_handlers_waiting_on_products`

File: PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    @objc public var fetching_offerings_error: String { "Error fetching offerings - %@" } //appleError
+    @objc public var fetching_products_failed: String { "SKRequest failed: %@" }  //appleError
+    @objc public var fetching_products_finished: String { "Products request finished" }  //debug
+    @objc public var fetching_products: String { "Requesting products from the store with identifiers: %@" } //debug
+    @objc public var found_existing_product_request: String { "Found an existing request for products: %@, appending to completion" }  //debug
+    @objc public var invalid_product_identifiers: String { "Invalid Product Identifiers - %@" } //appleWarning
+    @objc public var list_products: String { "%@ - %@" } //purchase
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift:20]: I bet we can do something smart here to be able to name parameters. not needed for this PR, but I'll give it some thought for the future. If this was swift-only it'd be pretty simple with an enum, but unfortunately obj-c compatibility makes things trickier

File: PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    @objc public var no_cached_offerings_fetching_network: String { "No cached Offerings, fetching from network" } //debug
+    @objc public var offerings_stale_updated_network: String { "Offerings updated from network." } // rcSuccess
+    @objc public var offerings_stale_updating_background: String { "Offerings cache is stale, updating from network in background" } //debug
+    @objc public var offerings_stale_updating_foreground: String { "Offerings cache is stale, updating from network in foreground" } //debug
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift:24]: `offerings_stale_updating_in_foreground`

File: PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    @objc public var products_already_cached: String { "Skipping products request because products were already cached. products: %@" } //debug
+    @objc public var retrived_products: String { "Retrived SKProducts: "  } //purchase
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift:26]: `retrieved_products`, there's a typo in the message as well (and probably in the original code)

File: PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    @objc public var starting_skproduct_request: String { "No existing requests and products not cached, starting SKProducts request for: %@" } //debug
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift:27]: `no_cached_requests_and_products_starting_skproduct_request`

Comment (aboedo) [reply PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift:27]: it's a long name, but it's better to have long named code than having messages the dev wouldn't expect. 
[iOS devs are quite used to long names](https://github.com/Quotation/LongestCocoa)

File: PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+//  PurchasesCoreSwift
+//
+//  Created by Tina Nguyen on 12/11/20.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCPurchaseStrings) public class PurchaseStrings: NSObject {
+    @objc public var cannot_purchase_product: String { "Could not purchase SKProduct. There is a problem with your configuration in App Store Connect. More info here: http://errors.rev.cat/products-empty" }  //appleWarning
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift:12]: `cannot_purchase_product_configuration_error` or even `cannot_purchase_product_appstore_configuration_error`

File: PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    @objc public var entitlements_revoked: String { "Entitlements revoked for product identifiers: %@. \nsyncing purchases" } //debug
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift:13]: `entitlements_revoked_syncing_purchases`, there should never be a message the developer doesn't expect when reading the class

File: PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    @objc public var finishing_transaction: String { "Finishing transaction %@ %@ (%@)" }  //purchase
+    @objc public var observer_active_finishtransaction_false: String { "Observer mode is active (finishTransactions is set to false) and purchase has been initiated. RevenueCat will not finish the transaction, are you sure you want to do this?" } //warning
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift:15]: `purchasing_with_observer_mode_and_finish_transactions_false_warning`

File: PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    @objc public var paymentqueue_removedtransaction: String { "PaymentQueue removedTransaction: %@ %@ (%@ %@) %@ - %d" }  //debug
+    @objc public var paymentqueue_revoke_entitlement: String { "PaymentQueue didRevokeEntitlementsForProductIdentifiers: %@" } //debug
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift:17]: `paymentqueue_revoked_entitlements_for_product_identifiers`

File: PurchasesCoreSwift/Logging/Strings/RestoreStrings.swift [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+//  PurchasesCoreSwift
+//
+//  Created by Tina Nguyen on 12/11/20.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCRestoreStrings) public class RestoreStrings: NSObject {
+    @objc public var sharing_acc_false_restore_called: String { "allowSharingAppStoreAccount is set to false and restoreTransactions has been called. Are you sure you want to do this?" } //warn
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/RestoreStrings.swift:12]: `restoretransactions_called_with_allow_sharing_appstore_account_false_warning`

Comment (aboedo) [reply PurchasesCoreSwift/Logging/Strings/RestoreStrings.swift:12]: abbreviating stuff like `acc` kinda hinders readability in favor of quicker writing

File: PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+//  PurchasesCoreSwift
+//
+//  Created by Tina Nguyen on 12/11/20.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCOfferingStrings) public class OfferingStrings: NSObject {
+    @objc public var cannot_find_product: String { "Could not find SKProduct for %@ \nThere is a problem with your configuration in App Store Connect. \nMore info here: http://errors.rev.cat/products-empty"} //appleWarning
```
[the remark this answers is not in the document]
Comment (tiinanguyen) [reply PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift:12]: note to self: update that link

File: PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    @objc public var completion_handler_waiting_on_products: String { "%lu completion handlers waiting on products" } //debug
+    @objc public var fetching_offerings_error: String { "Error fetching offerings - %@" } //appleError
+    @objc public var fetching_products_failed: String { "SKRequest failed: %@" }  //appleError
+    @objc public var fetching_products_finished: String { "Products request finished" }  //debug
+    @objc public var fetching_products: String { "Requesting products from the store with identifiers: %@" } //debug
+    @objc public var found_existing_product_request: String { "Found an existing request for products: %@, appending to completion" }  //debug
+    @objc public var invalid_product_identifiers: String { "Invalid Product Identifiers - %@" } //appleWarning
+    @objc public var list_products: String { "%@ - %@" } //purchase
+    @objc public var no_cached_offerings_fetching_network: String { "No cached Offerings, fetching from network" } //debug
+    @objc public var offerings_stale_updated_network: String { "Offerings updated from network." } // rcSuccess
```
Comment (tiinanguyen) [inline PurchasesCoreSwift/Logging/Strings/OfferingStrings.swift:22]: Note to self: offerings_updated_from_network

File: Purchases/Purchasing/RCStoreKitWrapper.m [hunk 14]
```diff
@@ -54,7 +54,7 @@ - (void)addPayment:(SKPayment *)payment {
 }
 
 - (void)finishTransaction:(SKPaymentTransaction *)transaction {
-    RCDebugLog(@"Finishing %@ %@ (%@)",
+    RCPurchaseLog(RCStrings.purchase.finishing_transaction,
                transaction.payment.productIdentifier,
                transaction.transactionIdentifier,
```
Comment (aboedo) [inline Purchases/Purchasing/RCStoreKitWrapper.m:59]: indentation is broken here

File: PurchasesCoreSwift/Logging/Strings/RestoreStrings.swift [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+//  PurchasesCoreSwift
+//
+//  Created by Tina Nguyen on 12/11/20.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCRestoreStrings) public class RestoreStrings: NSObject {
+    @objc public var restoretransactions_called_with_allow_sharing_appstore_account_false_warning: String { "allowSharingAppStoreAccount is set to false and restoreTransactions has been called. Are you sure you want to do this?" }
```
Comment (aboedo) [inline PurchasesCoreSwift/Logging/Strings/RestoreStrings.swift:12]: let's do 2 lines or more for long strings

Comment (aboedo) [reply PurchasesCoreSwift/Logging/Strings/RestoreStrings.swift:12]: char limit should be ~120

**Review by aboedo — APPROVED** [review]

**Pull request merged**